core/src/main/java/com/statful/client/core/api/MetricsSenderAPI.java

```diff
@@ -240,6 +240,20 @@ public void send() {
         }
     }
 
+    @Override
+    public void send(final long timestamp) {
```

Comment:
Instead of doing a method overload here, I think it'd be better to have the ability to define the timestamp in the fluent interface like every other metric field (name, value, etc.). That would allow to have a single send method which would check if `timestamp` is present and if not generate one at that time.